### PR TITLE
Add custom generation for sitemap

### PR DESCRIPTION
**Pull request by serg**
/assign @scheeles

### CODE REVIEW

File: config.toml [hunk 1]
```diff
@@ -2,6 +2,8 @@ baseURL = "https://docs.kubermatic.com/"
 theme = "hugo-theme-learn"
 languageCode = "en-us"
 title = "Kubermatic Docs"
+disableKinds = ["taxonomy"]
+ignoreErrors = ["error-disable-taxonomy"]
```
Comment (xrstf) [inline config.toml:6]: What's this about? Why do we need to ignore errors?

**Review by xrstf — COMMENTED** [review]
Questions

File: themes/hugo-theme-learn/layouts/_default/sitemap.xml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{{ printf "<?xml version=\"1.0\" encoding=\"utf-8\" standalone=\"yes\" ?>" | safeHTML }}
+<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml">
+{{- $products := .Site.Data.products -}}
+{{ range .Data.Pages }}
+{{- $isStableVersion := true -}}
```
Comment (xrstf) [inline themes/hugo-theme-learn/layouts/_default/sitemap.xml:5]: Why don't we generate the sitemap for _all_ versions?

Comment (serg) [reply themes/hugo-theme-learn/layouts/_default/sitemap.xml:5]: @xrstf this was decision of Sebastian.

File: config.toml (hunk 1 above)
Comment (serg) [reply config.toml:6]: @xrstf 
First line is prevent of generation for these useless (at current moment) pages:
https://docs.kubermatic.com/categories/
https://docs.kubermatic.com/tags/

Second line is needed for disable an error, that appears on build. Seems like Hugo expect taxonomy items by default. Also, some naming is changed for taxonomies from [v0.73](https://gohugo.io/news/0.73.0-relnotes/). The Hugo version was updated to 0.75 in this PR as well.

File: themes/hugo-theme-learn/layouts/_default/sitemap.xml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{{ printf "<?xml version=\"1.0\" encoding=\"utf-8\" standalone=\"yes\" ?>" | safeHTML }}
```
Comment (xrstf) [inline themes/hugo-theme-learn/layouts/_default/sitemap.xml:1]: Why is this a Go template line?

Comment (serg) [reply themes/hugo-theme-learn/layouts/_default/sitemap.xml:1]: I am just grab this [default template](https://gohugo.io/templates/sitemap-template/#hugos-sitemapxml) from recommendations on Hugo website. The [safeHTML](https://gohugo.io/functions/safehtml/) function is using for correct generation for HTML tags. There are attributes here with escaping for double quotes and they needs to be also generated correctly. Suppose, all above is also needed for XML tags as well.

**Comment by serg** [comment]
fix https://github.com/kubermatic/website/issues/1112

**Comment by serg** [comment]
@xrstf is it possible to merge this PR?

**Pull request merged**